lib/active_merchant/billing/gateways/braintree_blue.rb

```diff
@@ -537,12 +537,22 @@ def create_transaction_parameters(money, credit_card_or_vault_id, options)
             :first_name => credit_card_or_vault_id.first_name,
             :last_name => credit_card_or_vault_id.last_name
           )
-          parameters[:credit_card] = {
-            :number => credit_card_or_vault_id.number,
-            :cvv => credit_card_or_vault_id.verification_value,
-            :expiration_month => credit_card_or_vault_id.month.to_s.rjust(2, "0"),
-            :expiration_year => credit_card_or_vault_id.year.to_s
-          }
+          if credit_card_or_vault_id.is_a?(NetworkTokenizationCreditCard)
```

Comment:
> but this breaks when you have a string or a hash at the payment method (for example stripe token)

I don't quite follow here, but I'm the new kid on the block so that's to be expected.

![](http://cdn1.theodysseyonline.com/files/2015/01/26/635578417850183967-1833583414_new_kids320.jpg)

> Braintree is the only one I know of that specifically calls it apple pay

Cool. I'm fine leaving it as the generic "network tokenization" until a real-world scenario forces us to do otherwise.